x-pack/plugin/esql/qa/server/src/main/java/org/elasticsearch/xpack/esql/qa/rest/RestEsqlTestCase.java

```diff
@@ -343,38 +343,37 @@ public void testErrorMessageForInvalidTypeInParams() throws IOException {
     }
 
     public void testErrorMessageForLiteralDateMathOverflow() throws IOException {
-        List<String> datePeriodOverflowExpressions = List.of(
+        List<String> dateMathOverflowExpressions = List.of(
             "2147483647 day + 1 day",
             "306783378 week + 1 week",
-            "2147483647 year + 1 year"
-        );
-        // We cannot easily force an overflow using just milliseconds, since these are divided by 1000 and then the resulting seconds are
-        // stored in a long. But combining with seconds works.
-        List<String> timeDurationOverflowExpressions = List.of(
+            "2147483647 month + 1 month",
+            "2147483647 year + 1 year",
+            "-(-2147483647 year - 1 year)",
+            // We cannot easily force an overflow using just milliseconds, since these are divided by 1000 and then the resulting seconds
+            // are stored in a long. But combining with seconds works.
             "9223372036854775807 second + 1000 millisecond",
             "9223372036854775807 second + 1 second",
             "153722867280912930 minute + 1 minute",
             "2562047788015215 hour + 1 hour"
+
         );
 
-        for (String overflowExp : datePeriodOverflowExpressions) {
-            assertDateMathOverflow(overflowExp, "integer overflow");
-        }
-        for (String overflowExp : timeDurationOverflowExpressions) {
-            assertDateMathOverflow(overflowExp, "long overflow");
+        for (String overflowExp : dateMathOverflowExpressions) {
+            assertDateMathException(overflowExp, "overflow");
         }
+
+        assertDateMathException("-(-9223372036854775807 second - 1 second)", "Exceeds capacity of Duration");
     }
 
-    private static void assertDateMathOverflow(String overflowExpression, String expectedOverflowMessage) throws IOException {
+    private static void assertDateMathException(String overflowExpression, String errorSubstring) throws IOException {
         ResponseException re = expectThrows(
```

Comment:
Will this always be a `DateMathException` in these tests?

Reply:
Good point, probably not; when the folding behavior evolves, I don't know what exactly we'll have there, maybe a more general `ExpressionException`, `InvalidQueryException` or so?

I can generalize the test to expect a `EsqlClientException` instead, this way future readers will not think that is _has_ to be exactly this type of exception (for whatever reason).

Reply:
Actually, I just realized that `assertDateMathException` does not really check for a `DateMathException`, making the name misleading. I'll update the name.